### PR TITLE
NER support for Albert in run_ner.py and NerPipeline

### PR DESCRIPTION
**Pull request by marma**
I have added a class, AlbertForTokenClassification, based on BertForTokenClassification and added it to lists used for checking NER capabilities in run_ner.py and NerPipeline.

I tested NER fine-tuning on albert-base-v2, albert-large-v2 and albert-xlarge-v2 on the english CONLL 2003 dataset and they all get F1 of around 0.93-0.94, so it seems to be working. The fine-tuned models are published [here](https://huggingface.co/KB).

I've also added some command-line options to better control tokenization since different tokenizers have different possible arguments and defaults. I guess that in the end, when all tokenizers behave the same, these options will be unnecessary.

I changed how NerPipeline outputs tokens, from .decode(..) to .convert_ids_to_tokens(...) since it removes '_' at the beginning of tokens making it impossible (for sentencepiece tokens) to know which tokens form a word. Using .convert(...) would make sense if it were  outputting whole words and not words split into tokens. It might make sense to change this so that NerPipeline outputs whole words. That would assume that all the tokens in a word gets classified with the same label, which is not always the case.

I had one weird thing happening: when fine-tuning albert-large-v2 specifically for 3 or 4 epochs F1 would be reported as exactly 0. When setting num_train_epochs to 2 or 5 this did not happen. I'm going to assume that this has nothing to do with the code submitted :)

### CODE REVIEW

File: src/transformers/modeling_albert.py [hunk 1]
```diff
@@ -600,7 +600,7 @@ def forward(self, hidden_states):
         hidden_states = self.LayerNorm(hidden_states)
         hidden_states = self.decoder(hidden_states)
 
-        prediction_scores = hidden_states
+        prediction_scores = hidden_states + self.bias
```
Comment (LysandreJik) [inline src/transformers/modeling_albert.py]: This line was removed because it was a bug. Best to remove it here as well.

File: src/transformers/pipelines.py [hunk 2]
```diff
@@ -636,7 +636,7 @@ def __call__(self, *texts, **kwargs):
                 if self.model.config.id2label[label_idx] not in self.ignore_labels:
                     answer += [
                         {
-                            "word": self.tokenizer.decode([int(input_ids[idx])]),
+                            "word": self.tokenizer.convert_ids_to_tokens(int(input_ids[idx])),
```
Comment (LysandreJik) [inline src/transformers/pipelines.py]: Is there a reason you changed from `decode` to `convert_ids_to_tokens`?

**Review by LysandreJik — CHANGES_REQUESTED** [review]
Great addition, and nice job on getting high scores!! I've requested a few changes, it should be good to merge then.

File: src/transformers/pipelines.py (hunk 2 above)
Comment (marma) [reply src/transformers/pipelines.py]: Yes, I detailed it in the merge request :) 

In short, decode(...) tries to revert multiple tokens to the original word, but since tokens are iterated over on-by-one this makes no sense (or maybe I am missing something). Or rather, it does nothing for BERT, but it removes the leading '_' in Albert tokens which makes it impossible to reconstruct words from tokens since that conversion relies on the leading '_'.

Comment (LysandreJik) [reply src/transformers/pipelines.py]: Oops, indeed, I missed this! I'm asking @mfuntowicz who worked on the pipelines to verify :)

File: examples/ner/run_ner.py [hunk 3]
```diff
@@ -463,6 +469,22 @@ def main():
...
+    parser.add_argument(
+        "--keep_accents", action="store_const", const=True, help="Set this flag if model is trained with accents."
+    )
+
+    parser.add_argument(
+        "--strip_accents", action="store_const", const=True, help="Set this flag if model is trained without accents."
+    )
+
+    parser.add_argument(
+        "--nouse_fast",
```
Comment (mfuntowicz) [inline examples/ner/run_ner.py]: We agreed to use positive form on cli argument if possible _(as it avoid double negation)_. Can you rename this parameter `--use-fast` and update the logic accordingly ?

It also make sense for the core library, as we set use_fast to False by default in 2.5.1 to reduce the breaking changes introduced with tokenizers.

File: src/transformers/pipelines.py (hunk 2 above)
Comment (mfuntowicz) [reply src/transformers/pipelines.py]: Yap I agree on the behaviour you describe. It might change in the near future, as we were discussing to output the whole word NER class and not on a subtoken basis (as currently done). 

I'm fine with the proposed change.

**Review by mfuntowicz — CHANGES_REQUESTED** [review]

File: examples/ner/run_ner.py (hunk 3 above)
Comment (marma) [reply examples/ner/run_ner.py]: That makes sense. I'll fix it.

File: src/transformers/modeling_albert.py (hunk 1 above)
Comment (marma) [reply src/transformers/modeling_albert.py]: Fixed

**Review by LysandreJik — APPROVED** [review]
Cool, great that you added the tests!
